Code/Framework/AzCore/AzCore/Utils/Utils.cpp

```diff
@@ -410,6 +412,37 @@ namespace AZ::Utils
         return AZ::Success(AZStd::move(fileContent));
     }
 
+    AZ::Outcome<AZStd::string, AZStd::string> Get3rdPartyDirectory(AZ::SettingsRegistryInterface* settingsRegistry /*= nullptr*/)
+    {
+        // Locate the manifest file
+        auto manifestPath = GetO3deManifestPath(settingsRegistry);
+
+        // Read the content of the manifest file
+        const auto fileSize = AZ::IO::SystemFile::Length(manifestPath.c_str());
+        if (!fileSize)
+        {
+            return AZ::Failure(AZStd::string::format("Invalid manifest file %.*s", AZ_STRING_ARG(manifestPath)));
+        }
+        AZStd::vector<char> buffer(fileSize + 1);
+        buffer[fileSize] = '\0';
+        if (!AZ::IO::SystemFile::Read(manifestPath.c_str(), buffer.data()))
+        {
+            return AZ::Failure(AZStd::string::format("Error reading manifest file %.*s", AZ_STRING_ARG(manifestPath)));
+        }
+        const AZStd::string manifestContent = AZStd::string(buffer.begin(), buffer.end());
+
+        // Parse the manifest content and read the 'default_third_party_folder' value
+
+        rapidjson::Document configurationFile;
+        if (configurationFile.Parse(manifestContent.c_str()).HasParseError())
+        {
+            return AZ::Failure(AZStd::string::format("Error reading manifest file %.*s (JSON parse error)", AZ_STRING_ARG(manifestPath)));
+        }
```

Comment:
This could use the [AZ::JsonSerializationUtils::ReadJsonFile](https://github.com/o3de/o3de/blob/d3517fdf35be3a60a519b05d2e3e17ee338a98d3/Code/Framework/AzCore/AzCore/Serialization/Json/JsonUtils.cpp#L255-L277) function

Reply:
Thats much better